src/components/vm/vmMigrateDialog.jsx

```diff
@@ -0,0 +1,205 @@
+    );
+
+    if (copyDisabled) {
+        copyRadio = (
+            <Tooltip id="storage-copy-tooltip"
+                     content={_("Offline migration doesn't allow copying storage")}>
+                {copyRadio}
+            </Tooltip>
+        );
+    }
+
+    return (
+        <FormGroup label={_("Storage")}
+                   fieldId="storage"
+                   id="storage"
+                   hasNoPaddingTop>
+            <Radio id="shared"
+                   name="source"
+                   label={_("I have shared storage set up")}
+                   isChecked={storage === "shared"}
+                   onChange={() => setStorage("shared")} />
+            {copyRadio}
+        </FormGroup>
+    );
+};
+
+export const MigrateDialog = ({ vm, connectionName, toggleModal }) => {
+    const [destUri, setDestUri] = useState("");
+    const [error, setDialogError] = useState({});
+    const [inProgress, setInProgress] = useState(false);
+    const [storage, setStorage] = useState("shared");
+    const [temporary, setTemporary] = useState(false);
+    const [live, setLive] = useState(true);
+    const [validationFailed, setValidationFailed] = useState(false);
+
+    function validateParams() {
+        const validation = {};
+        if (isEmpty(destUri.trim()))
+            validation.destUri = _("Destination URI must not be empty");
+
+        return validation;
+    }
+
+    function onMigrate() {
+        if (!isObjectEmpty(validateParams())) {
+            setValidationFailed(true);
+            return;
+        }
+
+        setInProgress(true);
+        return migrateToUri(connectionName, vm.id, destUri, storage, live, temporary)
+                .then(toggleModal, exc => {
+                    setInProgress(false);
+                    setDialogError({ dialogError: _("Migration failed"), message: exc.message });
+                });
+    }
+
+    const body = (
+        <Form isHorizontal>
+            <DestUriRow destUri={destUri}
+                        setDestUri={setDestUri}
+                        validationFailed={validationFailed} />
+            <StorageRow storage={storage}
+                        setStorage={setStorage}
+                        live={live} />
+            <OptionsRow temporary={temporary}
+                        live={live}
+                        setStorage={setStorage}
+                        setTemporary={setTemporary}
+                        setLive={setLive} />
+        </Form>
+    );
+
+    const dataCorruptionWarning = () => {
+        const hasWriteableDisks = Object.values(vm.disks).findIndex(a => !a.readonly) !== -1;
+        if (storage === "shared" && temporary && hasWriteableDisks) {
+            return <Alert isInline
+                          variant="warning"
+                          id="data-corruption-warning"
+                          title={_("Do not run this VM on multiple hosts at the same time, as it would lead to data corruption.")} />;
```

Comment:
```suggestion
                          title={_("Do not run this VM on multiple hosts at the same time, as this will lead to data corruption.")} />;
```

Adjusting text a bit, including tense. (It's probably my mistake from quickly typing a suggestion on IRC, sorry.)